icu4c/source/test/cintltst/cloctst.c

```diff
@@ -6741,3 +6742,17 @@ static void TestBug20149() {
         log_err("ERROR: in uloc_forLanguageTag %s return %s\n", myErrorName(status), locale);
     }
 }
+
+// Test case for ICU-20575
+// This test checks if the environment variable LANG is set, 
+// and if so ensures that both C and C.UTF-8 cause ICU's default locale to be en_US_POSIX.
+static void TestCDefaultLocale(){
+    const char *defaultLocale = uloc_getDefault();
+    char *env_var = getenv("LANG");
+    if (env_var == NULL){
+      log_verbose("Skipping TestCDefaultLocale test, as the LANG variable is not set.");
+    }
+    else if( (strcmp(env_var, "C") == 0 || strcmp(env_var, "C.UTF-8") == 0) && strcmp(defaultLocale, "en_US_POSIX") != 0) {
+      log_err( "The default locale for LANG=C should be en-US_POSIX, not %s\n", defaultLocale );
```

Comment:
Hm, can we output the LANG variable here to check what it is?
(Right now this hard-codes LANG=C, but why not LANG=%s ?)